src/__tests__/useDebouncedValue.spec.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @jest-environment jsdom
+ */
+import { act, renderHook } from "@testing-library/react-hooks";
+import { useDebouncedValue } from "../hooks/useDebouncedValue";
+
+jest.useFakeTimers('modern');
+
+describe("useDebouncedValue", () => {
+  it("should be defined", () => {
+    expect(useDebouncedValue).toBeDefined();
+  });
+
+  it("should initialize with first value if timeout is not reached and initializeWithNull is false", () => {
+    const mockValue = "mock_value";
+
+    const { result } = renderHook(() => useDebouncedValue(mockValue, 200));
+
+    act(() => {
+      expect(result.current[0]).toBe(mockValue);
+    });
+  });
+
+
+  it("should return null if the timeout has not been reached and initializeWithNull is true", () => {
+    const mockValue = "mock_value";
+
+    const { result } = renderHook(() => useDebouncedValue(mockValue, 200, {initializeWithNull : true}));
+
+    act(() => {
+      expect(result.current[0]).toBeNull();
+    });
+  });
+
+  it("should returns updated value if the timeout has been reached and initializeWithNull is true", () => {
+    const mockValue = "mock_value";
+
+    const { result } = renderHook(() => useDebouncedValue(mockValue, 200, {initializeWithNull : true}));    
+    act(() => {
+      expect(result.current[0]).toBeNull();
+      jest.runAllTimers();
+      expect(result.current[0]).toBe(mockValue);
+    });
+  });
+
+  it("should respect the timeout value if initializedWithNull is true", () => {
+    const mockValue = "mock_value";
+    const mockTimeout = 200;
+
+    const { result } = renderHook(() =>
+      useDebouncedValue(mockValue, mockTimeout, {initializeWithNull : true})
+    );
+
+    act(() => {
+      jest.advanceTimersByTime(mockTimeout - 5);
+      expect(result.current[0]).toBe(null);
+    });
+
+    act(() => {
+      // accounting for timing issues as it's not exactly accurate
+      jest.advanceTimersByTime(10);
```

Comment:
I couldn't figure this out. Trying it with `mockTimeout - 1` and  `1` didn't help me. Can you take a look here @qiweiii because you had already fixed it on your end in the previous `HEAD`

Reply:
I think it might be because of a slight lag in `useEffect` not entirely sure. But `mockTimeout - x` and `x+1` work but not   `mockTimeout - x` and `x` 🤔

Reply:
@imbhargav5 move `expect` out of `act` will work